project-chip/connectedhomeip
# Add FailRandomlyAtFault command to Fault Injection cluster

#### Problem
What is being fixed?  Examples:
*  Missing FailRandomlyAtFault command in Fault Injection cluster,  this command allow test harness to configure a fault to be triggered randomly, with a given probability defined as a percentage, this is meant to be used on live systems to generate a build that will encounter random failures.

* Fixes #21941

#### Change overview
Add FailRandomlyAtFault command to Fault Injection cluster

#### Testing
How was this tested? (at least one bullet point required)
* Build all-clusters-app with nlfaultinjection enabled
`scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug/standalone chip_config_network_layer_ble=false chip_with_nlfaultinjection=true`

* Configure a system fault from chip_tool
`./chip-tool faultinjection fail-randomly-at-fault 1 1 45 12344321 0`

* Confirm the fault is activated from all-clusters-app log
```
[1660840083.976320][3208798:3208798] CHIP:EM: Received message of type 0x8 with protocolId (0, 1) and MessageCounter:110067918 on exchange 46251r
[1660840083.976342][3208798:3208798] CHIP:EM: Handling via exchange: 46251r, Delegate: 0x5599f60e2928
[1660840083.976382][3208798:3208798] CHIP:DMG: InvokeRequestMessage =
[1660840083.976395][3208798:3208798] CHIP:DMG: {
[1660840083.976409][3208798:3208798] CHIP:DMG: 	suppressResponse = false, 
[1660840083.976422][3208798:3208798] CHIP:DMG: 	timedRequest = false, 
[1660840083.976435][3208798:3208798] CHIP:DMG: 	InvokeRequests =
[1660840083.976454][3208798:3208798] CHIP:DMG: 	[
[1660840083.976466][3208798:3208798] CHIP:DMG: 		CommandDataIB =
[1660840083.976480][3208798:3208798] CHIP:DMG: 		{
[1660840083.976492][3208798:3208798] CHIP:DMG: 			CommandPathIB =
[1660840083.976506][3208798:3208798] CHIP:DMG: 			{
[1660840083.976520][3208798:3208798] CHIP:DMG: 				EndpointId = 0x0,
[1660840083.976534][3208798:3208798] CHIP:DMG: 				ClusterId = 0xfff1fc06,
[1660840083.976548][3208798:3208798] CHIP:DMG: 				CommandId = 0x1,
[1660840083.976560][3208798:3208798] CHIP:DMG: 			},
[1660840083.976575][3208798:3208798] CHIP:DMG: 			
[1660840083.976588][3208798:3208798] CHIP:DMG: 			CommandFields = 
[1660840083.976602][3208798:3208798] CHIP:DMG: 			{
[1660840083.976617][3208798:3208798] CHIP:DMG: 				0x0 = 1, 
[1660840083.976631][3208798:3208798] CHIP:DMG: 				0x1 = 1, 
[1660840083.976645][3208798:3208798] CHIP:DMG: 				0x2 = 45, 
[1660840083.976659][3208798:3208798] CHIP:DMG: 			},
[1660840083.976671][3208798:3208798] CHIP:DMG: 		},
[1660840083.976689][3208798:3208798] CHIP:DMG: 		
[1660840083.976700][3208798:3208798] CHIP:DMG: 	],
[1660840083.976718][3208798:3208798] CHIP:DMG: 	
[1660840083.976730][3208798:3208798] CHIP:DMG: 	InteractionModelRevision = 1
[1660840083.976742][3208798:3208798] CHIP:DMG: },
[1660840083.976788][3208798:3208798] CHIP:DMG: AccessControl: checking f=1 a=c s=0x000000000001B669 t= c=0xFFF1_FC06 e=0 p=m
[1660840083.976808][3208798:3208798] CHIP:DMG: AccessControl: allowed
[1660840083.976824][3208798:3208798] CHIP:DMG: Received command for Endpoint=0 Cluster=0xFFF1_FC06 Command=0x0000_0001
[1660840083.976845][3208798:3208798] CHIP:ZCL: FaultInjection: Configure a fault of type: 1 and Id: 1 to be triggered randomly

```
